### PR TITLE
Ca feature/change port to 9000 and add some changes

### PR DESCRIPTION
**Pull request by cagodoy**
# Relations

```bash

                                                     +------------+
                                                     |            |
 +---------+                  +--------------------->+ Criterias  |
 |         |                  |         1 : 1        |            |
 |  Users  |                  |                      +------------+
 |         |                  |
 +----+----+                  |
      ^             +---------+--------+             +------------+
      |             |                  |    1 : 1    |            |
      |             | SubjectsChannels +<----------->+  Channels  |
      |             |                  |             |            |
      |             +---------+--------+             +------------+
      |                       ^
      |                       | 1 : N
      |                       v
      |                +------+-----+                +------------+
      |    1 : N       |            |     1 : N      |            |
      +--------------->+  Subjects  +<-------------->+  Messages  |
                       |            |                |            |
                       +------------+                +------------+

```
- [X] Define statuses (ca)
- [X] Implement status behavior on message lifecycle (ca)
- [X] Define and implements mocks struct models (ca)
- [X] Implement mock to `GET /api/v1/subject` endpoint (ca)
- [X] Implement `GET /api/v1/messages/:id/status` endpoint
- [X] Define and implement `pigeon-http` channel (ca)
- [X] Add `pigeon-http` value to subject model {options: {headers}} (ca)
- [X] Append user-channels `options` to content message by channel (ca)
- [X] Add 'data.json' on docker (ca)
- [X] Implement `Messages` table on `BoltDB` (ca)
- [X] Add Status on `Messages` protobuf (ca)
- [X] Implement logic to validate and get user by x-api-key header value (ca)
- [X] Add user_id to message protobuf and implement this in put scheduler method (ca)
- [X] Add criteria model (ca)
- [X] Add criteria examples to mock (ca)
- [X] Implement criteria when create new message (ca)

### CODE REVIEW

File: cmd/scheduler/main.go [hunk 1]
```diff
@@ -46,7 +71,5 @@ func main() {
 	reflection.Register(s)
 	if err := s.Serve(lis); err != nil {
 		log.Fatal(err)
-	} else {
-		fmt.Println("Runing and ready bitches!")
```
Comment (mtavano) [inline cmd/scheduler/main.go:50]: best log ever

File: db/criterias.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		if value.ID == ID {
+			return value, nil
+		}
+	}
+
+	return nil, errors.New("criteria not found")
+}
+
+// GetCriteriaDelay ...
+func (ts *CriteriaStore) GetCriteriaDelay(criteriaID string, criteriaCustom int64) (time.Duration, error) {
```
Comment (mtavano) [inline db/criterias.go:42]: could u explain this method to me?? I think I'm not catching the main idea behind this

File: cmd/scheduler/main.go (hunk 1 above)
Comment (cagodoy) [reply cmd/scheduler/main.go:50]: @mtavano ready for merge PR?

File: db/users.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+// GetUserByAPIKey ...
+func (us *UserStore) GetUserByAPIKey(APIKey string) (*pigeon.User, error) {
+	mock, err := getMock()
+	if err != nil {
+		return nil, err
+	}
+
+	var user *pigeon.User
```
Comment (mtavano) [inline db/users.go:31]: do not use literal.

```go
user := new(pigeon.User)
```

File: db/users.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+	for _, value := range mock.Users {
+		if value.APIKey == APIKey {
+			user = value
+			continue
```
Comment (mtavano) [inline db/users.go:36]: this line should be `break`

**Review by mtavano — APPROVED** [review]

**Pull request merged**